src/lib/components/EventNetwork.svelte

```diff
@@ -27,29 +34,215 @@
   interface NetworkNode extends d3.SimulationNodeDatum {
     id: string;
     event?: NDKEvent;
-    index?: number;
+    level: number;
     isContainer: boolean;
     title: string;
     content: string;
     author: string;
     type: "Index" | "Content";
+    naddr?: string;
+    nevent?: string;
     x?: number;
     y?: number;
-    fx?: number | null;
-    fy?: number | null;
-    vx?: number;
-    vy?: number;
   }
 
   interface NetworkLink extends d3.SimulationLinkDatum<NetworkNode> {
     source: NetworkNode;
     target: NetworkNode;
     isSequential: boolean;
   }
+  interface GraphData {
+    nodes: NetworkNode[];
+    links: NetworkLink[];
+  }
+  interface GraphState {
+    nodeMap: Map<string, NetworkNode>;
+    links: NetworkLink[];
+    eventMap: Map<string, NDKEvent>;
+    referencedIds: Set<string>;
+  }
+  function initializeGraphState(events: NDKEvent[]): GraphState {
```

Comment:
Just for easier readability, I'd recommend putting a blank line between declarations.

```suggestion
  interface GraphData {
    nodes: NetworkNode[];
    links: NetworkLink[];
  }
  
  interface GraphState {
    nodeMap: Map<string, NetworkNode>;
    links: NetworkLink[];
    eventMap: Map<string, NDKEvent>;
    referencedIds: Set<string>;
  }
  
  function initializeGraphState(events: NDKEvent[]): GraphState {
```

Reply:
I was wondering about code conventions, yesterday. Do you have any, that you could share with us?

Reply:
Hmm I can probably dredge up a few.  I'll try to focus on the meaningful, rather than semantic.  I can put together a small style guide for this repo.

I'll likely need to do the same for C++ for Aedile, as we develop a consistent style.